src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,228 @@
+                    index = hash(nextNode.key);
+                    //new linked list is empty
+                    if (newTable[index] == null) {
+                        newTable[index] = nextNode;
+                        //what should we do if not
+                    } else {
+                        //does it work correctly?
+                        //seems yes
+                        //newTable[index].next = nextNode;
+                    }
+                    node = node.next;
+                }
+            }
+
+        }
+
+        table = newTable;
+        threshold = (int) (CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
+
+//    private void putPair(K key, V value) {
+//        int index = hash(key);
+//        Node<K, V> newNode = new Node<>(key, value, null);
+//        Node<K, V> node = table[index];
+//        //linked list is empty
+//        if (node == null) {
+//            table[index] = newNode;
+//            size++;
+//            //linked list isn't empty
+//        } else {
+//            //оно не заходит сюда
+//            //node.key.equals(key)
+//            while (node.next != null || checkKey(key, node)) {
+//                if (checkKey(key, node)) {
+//                    node.value = value;
+//                    return;
+//                }
+//                node = node.next;
+//            }
+//            //okay if we don't have the same keys -> we should put new pair
+//            node.next = newNode;
+//            size++;
+//        }
+//    }
+
+    private boolean checkKey(K key, Node<K, V> node) {
+        if (node.key == key || node.key != null && node.key.equals(key)) {
+            return true;
+        }
+        return false;
+    }
+
+
+    static class Node<K, V> implements Map.Entry<K, V> {
+        final K key;
+        V value;
+        Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        @Override
+        public K getKey() {
+            return key;
+        }
+
+        @Override
+        public V getValue() {
+            return value;
+        }
+
+        @Override
+        public V setValue(V newValue) {
+            V oldValue = value;
+            value = newValue;
+            return oldValue;
+        }
+
+        @Override
+        public final int hashCode() {
+//            Random random = new Random();
+//            // int result = random.nextInt(1000);
+//            //int number = random.nextInt(1000);
+            // int hash = 88;
+            //if it won't work you will try without verify on null
+            //and without value
+//            hash = 35 * hash + (key == null ? 0 : key.hashCode());
+//            hash = 35 * hash + (value == null ? 0 : value.hashCode());
+//            return hash;
+            return Objects.hashCode(key) ^ Objects.hashCode(value);
+        }
+
+
+        public static int hash(Object key) {
```

Comment:
The `hash` method should not be public or static. It should be private and non-static. Also, you should not use CAPACITY directly in the `hash` method, as the capacity may change when the table is resized. Pass the capacity as a parameter to the method instead.